plugin-server/src/main/ingestion-queues/session-recording/services/session-manager-v3.ts

```diff
@@ -133,26 +139,70 @@ export class SessionManagerV3 {
     private async setup(): Promise<void> {
         await mkdir(this.context.dir, { recursive: true })
 
+        const bufferFileExists = await stat(this.file(BUFFER_FILE_NAME))
+            .then(() => true)
+            .catch(() => false)
+
+        let metadataFileContent: string | undefined
+        let context: SessionManagerBufferContext | undefined
+
+        if (!bufferFileExists) {
+            return
+        }
+
         try {
-            const fileExists = await stat(this.file('metadata.json')).then(
-                () => true,
-                () => false
-            )
-            if (fileExists) {
-                const bufferMetadata: SessionManagerBufferContext = JSON.parse(
-                    await readFile(this.file('metadata.json'), 'utf-8')
-                )
-                this.buffer = {
-                    context: bufferMetadata,
-                    fileStream: this.createFileStreamFor(path.join(this.context.dir, BUFFER_FILE_NAME)),
-                }
-            }
+            metadataFileContent = await readFile(this.file(METADATA_FILE_NAME), 'utf-8')
+            context = JSON.parse(metadataFileContent)
         } catch (error) {
```

Comment:
i guess since efs is reading over the network we could have transient failures to read so should maybe have a retry in future but 🤷

Reply:
I _thought_ that isn't possible with EFS in that the mounting thing takes care of that for us...

Reply:
But tbh I am also mostly 🤷